File: back-spring-boot/src/main/java/com/groupe2/backspringboot/model/Entreprises.java
package com.groupe2.backspringboot.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Version;

import lombok.Data;

@Data
@Entity
@Table(name = "Entreprises")
public class Entreprises {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	public int id;
	public String entreprise;
	public String adresse;
	public String departement;
	public String telephone;
	public String mail;
	public String profession;
	
	@Version
	private int version;
	
}

File: front-angular/src/app/entreprise/entreprises.ts
export class Entreprises {
    id: number;
    adresse: string;
    departement: string;
    entreprise: string;
    mail: string;
    profession: string;
    telephone: string;
    version: number;

    constructor(id: number, adresse: string, departement: string, entreprise: string, mail: string, profession: string, telephone: string,
        version: number) {
        this.id = id;
        this.adresse = adresse;
        this.departement = departement;
        this.entreprise = entreprise;
        this.mail = mail;
        this.profession = profession;
        this.telephone = telephone;
        this.version = version;
    }

}
